plugins/woocommerce/src/Blocks/BlockTypes/AbstractBlock.php

```diff
@@ -95,6 +114,7 @@ public function render_callback( $attributes = [], $content = '', $block = null
 			$this->register_block_type_assets();
 			$this->enqueue_assets( $render_callback_attributes, $content, $block );
 		}
+		$this->interactivity_api_config->initialize_shared_config();
```

Comment:
Are we 100% sure that it is necessary to serialize the default settings every time there is a block in the front end, whatever that block may be?

Reply:
We are not sure of that no. We can remove this and get blocks to explicitly do this instead.

Reply:
It would still be good if there were a global way to serialize those settings if it's something that multiple blocks need to do. But perhaps we could combine it with the same requirement we have for the state:

https://github.com/woocommerce/woocommerce/blob/381d06e0cd5368938c5f61c0e28ba0a763fb8ec7/plugins/woocommerce/src/Blocks/BlockTypes/ProductButton.php#L293-L309

Reply:
@luisherranz I've moved over into a new PR and opted to explicitly init when needed. if multiple blocks call it still happens just once, but I wonder if you're looking for something else, we can discuss there https://github.com/woocommerce/woocommerce/pull/57469/files#diff-cdfe42b347ee28e297ddff155cd570f732c92f9f1b07371bc09a1eb83fccc78bR43